File: dataset/output-bk/MISTRAL_codestral-latest/Task199_MISTRAL_codestral-latest.cpp
#include <iostream>
#include <iomanip>

class Clock {
private:
    int minutes;

public:
    Clock(int hours, int minutes) {
        this->minutes = (hours * 60 + minutes) % (24 * 60);
    }

    bool operator==(const Clock& other) const {
        return this->minutes == other.minutes;
    }

    Clock operator+(int minutes) const {
        return Clock(0, this->minutes + minutes);
    }

    Clock operator-(int minutes) const {
        return Clock(0, this->minutes - minutes);
    }

    friend std::ostream& operator<<(std::ostream& os, const Clock& clock) {
        int hours = clock.minutes / 60;
        int minutes = clock.minutes % 60;
        os << std::setfill('0') << std::setw(2) << hours << ":" << std::setw(2) << minutes;
        return os;
    }
};

File: dataset/output-bk/MISTRAL_codestral-latest/Task199_MISTRAL_codestral-latest.java
public class Clock {
    private int minutes;

    public Clock(int hours, int minutes) {
        this.minutes = (hours * 60 + minutes) % (24 * 60);
    }

    public boolean equals(Clock other) {
        return this.minutes == other.minutes;
    }

    public Clock add(int minutes) {
        return new Clock(0, this.minutes + minutes);
    }

    public Clock subtract(int minutes) {
        return new Clock(0, this.minutes - minutes);
    }

    @Override
    public String toString() {
        int hours = this.minutes / 60;
        int minutes = this.minutes % 60;
        return String.format("%02d:%02d", hours, minutes);
    }
}